web/src/main/java/com/microsoft/applicationinsights/web/internal/WebAppInitializer.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.web.internal;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collection;
+
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContext;
+import javax.servlet.FilterRegistration;
+
+import com.microsoft.applicationinsights.internal.agent.AgentConnector;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+import com.google.common.base.Strings;
+
+/**
+ * Created by gupele on 5/7/2015.
+ */
+//@WebListener
+public final class WebAppInitializer implements ServletContextListener {
+
+    private final static String FILTER_NAME = "ApplicationInsightsWebFilter";
+    private final static String WEB_INF_FOLDER = "WEB-INF/";
+
+    private String name;
+
+    /**
+     * The method is called by the container before the WebApp is initialized
+     * The method will fetch the WebApp name and then will register itself with that name
+     * The registration should return a 'key' which is later used to identify threads
+     * that are originating from this WebApp. To do so, the class will create a filter
+     * and pass the 'key' to that filter, the filter is responsible for assigning the 'key' to each thread.
+     *
+     * Note, that the method will take care of situations where the filter is already declared in the 'web.xml'
+     *
+     * @param sce The Servlet Context Event that we use to extract the data we need.
+     */
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+        try {
+            ServletContext context = sce.getServletContext();
+
+            name = getName(context);
+
+            String key = registerWebApp(name);
+
+            addFilter(context);
```

Comment:
I think it would make the code more readable if the call to `addFilter` would appear just after getting the context as this way all the other commands related to the registration would appear , i.e.:

    ServletContext context = sce.getServletContext();
    addFilter(context);
    name = getName(context);
    String key = registerWebApp(name);
    WebRequestTrackingFilter.setName(key);